src/OpenLoco/Vehicles/VehicleHead.cpp

```diff
@@ -3245,6 +3246,33 @@ namespace OpenLoco::Vehicles
             }
         }
 
+        return cargoLUTToString(cargoTotals, buffer);
+    }
+
+    char* VehicleHead::generateCargoCapacityString(char* buffer)
+    {
+        std::array<uint32_t, ObjectManager::getMaxObjects(ObjectType::cargo)> cargoTotals{};
+        Vehicles::Vehicle train(this);
+        for (auto& car : train.cars)
+        {
+            auto front = car.front;
+            auto body = car.body;
+
+            if (front->secondaryCargo.type != 0xFF)
+            {
+                cargoTotals[front->secondaryCargo.type] += front->secondaryCargo.maxQty;
+            }
+            if (body->primaryCargo.type != 0xFF)
+            {
+                cargoTotals[body->primaryCargo.type] += body->primaryCargo.maxQty;
+            }
+        }
+
+        return cargoLUTToString(cargoTotals, buffer);
+    }
+
+    char* VehicleHead::cargoLUTToString(std::array<uint32_t, 32>& cargoTotals, char* buffer)
```

Comment:
You've switched the declaration to use ObjectManager::getMaxObjects(ObjectType::cargo) but not the definition. Can you keep it consistent.

Reply:
sorry I missed this reference, do you mind if I alias the type out

Reply:
something like `using CargoTotalArray = std::array<uint32_t, ObjectManager::getMaxObjects(ObjectType::cargo)>;`

Reply:
Sure.

Reply:
done